azurerm/internal/services/compute/windows_virtual_machine_resource.go

```diff
@@ -0,0 +1,1004 @@
+				ConflictsWith: []string{
+					// TODO: "virtual_machine_scale_set_id"
+					"zone",
+				},
+			},
+
+			"boot_diagnostics": bootDiagnosticsSchema(),
+
+			"computer_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+
+				// Computed since we reuse the VM name if one's not specified
+				Computed: true,
+				ForceNew: true,
+				// note: whilst the portal says 1-15 characters it seems to mirror the rules for the vm name
+				// (e.g. 1-15 for Windows, 1-63 for Windows)
+				ValidateFunc: ValidateWindowsName,
+			},
+
+			"custom_data": base64.OptionalSchema(true),
+
+			"dedicated_host_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true, // TODO: investigate, looks like the Portal allows migration
+				ValidateFunc: computeValidate.DedicatedHostID,
+				// the Compute/VM API is broken and returns the Resource Group name in UPPERCASE :shrug:
+				DiffSuppressFunc: suppress.CaseDifference,
+				// TODO: raise a GH issue for the broken API
+				// /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/TOM-MANUAL/providers/Microsoft.Compute/hostGroups/tom-hostgroup/hosts/tom-manual-host
+			},
+
+			"enable_automatic_updates": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true, // TODO: confirm
+				Default:  true,
+			},
+
+			"eviction_policy": {
+				// only applicable when `priority` is set to `Spot`
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					// NOTE: whilst Delete is an option here, it's only applicable for VMSS
+					string(compute.Deallocate),
+				}, false),
+			},
+
+			"identity": virtualMachineIdentitySchema(),
+
+			"license_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"None",
+					"Windows_Client",
+					"Windows_Server",
+				}, false),
+			},
+
+			"max_bid_price": {
+				Type:     schema.TypeFloat,
+				Optional: true,
+				Default:  -1,
+			},
+
+			"plan": planSchema(),
+
+			"priority": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  string(compute.Regular),
+				ValidateFunc: validation.StringInSlice([]string{
+					string(compute.Regular),
+					string(compute.Spot),
+				}, false),
+			},
+
+			"provision_vm_agent": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+				ForceNew: true,
+			},
+
+			"proximity_placement_group_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: computeValidate.ProximityPlacementGroupID,
+				// the Compute/VM API is broken and returns the Resource Group name in UPPERCASE :shrug:
```

Comment:
I'd say lets open a SDK issue, but my last one was closed by the compute team: https://github.com/Azure/azure-sdk-for-go/issues/5699

Reply:
I'll request the issue's re-opened on the Swagger repo, which should ensure the bugs tracked; then we can escalate via internal channels